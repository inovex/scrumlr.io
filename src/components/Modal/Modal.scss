@import '../../constants/style';

.modal__content-wrapper {
  overflow: auto;
  //margin-top: $large-margin;

  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100px;

  width: 100%;
  height: 100%;
}

.modal__content {
  position: relative;
  background: $color-white;
  min-width: 200px;
  max-width: 860px;
  min-height: 100px;
  padding: $large-padding * 1.5;
  box-shadow: 0px 32px 64px 0 rgba(0, 0, 0, 0.75);
  margin: 64px;
  border-radius: 6px;
}

.modal__close-button {
  position: absolute;
  top: $large-margin;
  right: $large-margin;

  background: none;
  margin: 0;
  padding: 0;
  outline: none;
  border: none;

  cursor: pointer;
}

.modal__close-button * {
  transition: fill .15s ease-in-out;
  fill: $color-dark-gray;
}

.modal__close-button:focus *,
.modal__close-button:hover * {
  fill: $color-mint;
}

.modal__headline {
  align-self: flex-start;
  margin-top: $default-margin;
}

.modal__action-area {
  display: flex;
  justify-content: flex-end;

  margin-top: $large-margin
}

.modal__ack-button {
  align-self: flex-end;

  border: 0;
  background: none;
  outline: none;

  font-size: $theme-font-size;
  font-weight: bold;
  color: $color-mint;

  padding: $default-padding;
  transition: background .2s ease-in-out, color .15s ease-in-out;
}

.modal__ack-button:focus,
.modal__ack-button:hover {
  color: $color-white;
  background: $color-mint;
}