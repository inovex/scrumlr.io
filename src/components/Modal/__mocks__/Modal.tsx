import * as React from 'react';

export class Modal extends React.Component<any, {}> {
  render() {
    return <div>{this.props.children}</div>;
  }
}

export default Modal;
