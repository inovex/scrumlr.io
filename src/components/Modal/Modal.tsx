import * as React from 'react';

import './Modal.scss';
import { mapDispatchToProps } from './Modal.container';
import { connect } from 'react-redux';
import Portal from '../Portal';

export interface OwnModalProps {
  onClose?: () => void;
  onSubmit?: () => void;
  children: any;
}

export interface DispatchModalProps {
  onStatus: (open: boolean) => void;
}

export type ModalProps = OwnModalProps & DispatchModalProps;

export class Modal extends React.Component<ModalProps, {}> {
  componentWillMount() {
    this.props.onStatus(true);
  }

  componentWillUnmount() {
    this.props.onStatus(false);
  }

  render() {
    return (
      <Portal onClose={this.props.onClose}>
        <div className="modal__content-wrapper">
          <div className="modal__content">
            {this.props.children}
            {this.props.onSubmit && (
              <div className="modal__action-area">
                <button
                  type="button"
                  onClick={this.props.onSubmit}
                  className="modal__ack-button"
                >
                  OK
                </button>
              </div>
            )}
          </div>
        </div>
      </Portal>
    );
  }
}

export default connect<{}, DispatchModalProps, OwnModalProps>(
  null,
  mapDispatchToProps
)(Modal);
