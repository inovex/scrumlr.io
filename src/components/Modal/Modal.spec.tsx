import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Modal, ModalProps } from './Modal';

describe('<Modal />', () => {
  describe('dumb component', () => {
    let props: ModalProps;
    let shallowWrapper: ShallowWrapper<ModalProps, {}>;

    beforeEach(() => {
      props = {
        onClose: jest.fn(),
        onSubmit: jest.fn(),
        onStatus: jest.fn(),
        children: 'Test'
      };

      shallowWrapper = shallow(<Modal {...props} />);
    });

    describe('rendering', () => {
      it('should match the snapshot', () => {
        expect(shallowWrapper.debug()).toMatchSnapshot();
      });

      it('should call onStatus on mount', () => {
        expect(props.onStatus).toHaveBeenCalled();
      });

      it('should call onSubmit on click on OK button', () => {
        expect(props.onSubmit).not.toHaveBeenCalled();
        shallowWrapper.find('.modal__ack-button').simulate('click');
        expect(props.onSubmit).toHaveBeenCalled();
      });
    });
  });
});
