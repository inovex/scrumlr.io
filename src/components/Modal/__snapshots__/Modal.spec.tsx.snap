// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> dumb component rendering should match the snapshot 1`] = `
"<div className=\\"modal modal__backdrop\\">
  <FocusLock disabled={false} returnFocus={false} noFocusGuards={false} autoFocus={true} persistentFocus={false} allowTextSelection={[undefined]} group={[undefined]} className={[undefined]} whiteList={[undefined]} as=\\"div\\" lockProps={{...}} onActivation={[undefined]} onDeactivation={[undefined]}>
    <div className=\\"modal__content-wrapper\\">
      <div className=\\"modal__content\\">
        Test
        <div className=\\"modal__action-area\\">
          <button className=\\"modal__close-button\\" type=\\"button\\" onClick={[Function: mockConstructor]}>
            <mockConstructor name=\\"close-circle\\" width={48} height={48} />
          </button>
          <button type=\\"button\\" onClick={[Function: mockConstructor]} className=\\"modal__ack-button\\">
            OK
          </button>
        </div>
      </div>
    </div>
  </FocusLock>
</div>"
`;
